File: java/src/test/java/UnsubscribeServiceTest.java
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.mockito.Mockito.*;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.Mock;


public class UnsubscribeServiceTest {

    private List<HashMap<String, String>> statementCharges;
    private UnsubscribeService service;

    @BeforeEach
    public void beforeFunction() {
        service = mock(UnsubscribeService.class);
        statementCharges = new ArrayList<HashMap<String, String>>();

        when(service.displayRecurringCharges()).thenCallRealMethod();
        when(service.getCharges()).thenReturn(statementCharges);
    }

    @Test
    public void testDoesNotIncludeChargesThatOnlyOccurOnce() {
        addCharge("Spotify", "1/1/2019", "9.99");

        assertFalse(service.displayRecurringCharges().containsKey("Spotify"));
    }

    @Test
    public void testSumsTheChargesThatOccurMoreThanOnce() {
        addCharge("Spotify", "1/1/2019", "9.99");
        addCharge("Spotify", "2/1/2019", "9.99");

        assertTrue(2 == service.displayRecurringCharges().get("Spotify"));
    }

    private void addCharge(String name, String date, String amount) {
        HashMap<String, String> statementCharge = new HashMap<String, String>();
        statementCharge.put("name", name);
        statementCharge.put("date", date);
        statementCharge.put("amount", amount);
        statementCharges.add(statementCharge);
    }
}

File: elixir/test/unsubscribe_service_test.exs
defmodule UnsubscribeServiceTest do
  use ExUnit.Case

  test "sums the charges that occur more than once" do
    sample_name = "Spotify"

    charges = [
      %{amount: 9.99, date: "2/1/2018", name: sample_name},
      %{amount: 9.99, date: "3/1/2018", name: sample_name}
    ]

    assert UnsubscribeService.display_recurring_charges(charges)[sample_name] == 2
  end

  test "does not include charges that only occur once" do
    sample_name = "Spotify"
    charges = [%{amount: 9.99, date: "2/1/2018", name: sample_name}]

    assert UnsubscribeService.display_recurring_charges(charges)[sample_name] == nil
  end
end
